channel.go

```diff
@@ -563,7 +569,13 @@ func (ch *Channel) Connect(ctx context.Context, hostPort string) (*Connection, e
 	}
 
 	timeout := getTimeout(ctx)
-	tcpConn, err := dialContext(ctx, hostPort)
+	var tcpConn net.Conn
+	var err error
+	if ch.dialer != nil {
```

Comment:
instead of an `if` here, can we set the dialer function to `dialContext` in the constructor instead. That way we avoid some of the noise here and simplify this logic.